server.js

```diff
@@ -1,35 +1,197 @@
+        ...nominee,
+        // eslint-disable-next-line max-len
+        category: categoriesArray.find((oneCategory) => oneCategory.description === nominee.category)
+      })
+      await newNominee.save()
+    })
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app));
+})
+
+app.get('/categories', async (req, res) => {
+  const categories = await Category.find()
+  res.json(categories)
+})
+
+app.get('/categories/:categoryId/nominees', async (req, res) => {
+  const { categoryId } = req.params
+  const { awardYear, nominee, film, win } = req.query
+  
+  try {
+    const query = {}
+    query.category = categoryId
+    if (awardYear) {
+      query.year_award = awardYear
+    }
+    if (nominee) {
+      query.nominee = { $regex: new RegExp(nominee, "i") }
+    }
+    if (film) {
+      query.film = { $regex: new RegExp(film, "i") }
+    }
+    if (win) {
+      query.win = win
+    }
+    
+    const nomineesCategory = await Nominee.find(query).populate('category')
+    if (nomineesCategory) {
+      res.json(nomineesCategory);
+    } else {
+      res.status(404).json({ error: "Not found" })
+    }
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+});
+
+app.get('/nominees', async (req, res) => {
+  const { awardYear, nominee, page, perPage } = req.query
+  try {
+    const query = {}
+    if (awardYear) {
+      query.year_award = awardYear
+    }
+    if (nominee) {
+      query.nominee = { $regex: new RegExp(nominee, "i") }
+    }
+    const nominees = await Nominee.find(query).populate('category').skip(((Number(page) - 1) * Number(perPage))).limit(Number(perPage))
+    res.json(nominees);
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+})
+
+app.get('/nominees/:id', async (req, res) => {
+  const { id } = req.params
+  try {  
+    const nominee = await Nominee.findById(id)
+    if (nominee) {
+      res.json(nominee)
+    } else {
+      res.status(404).json({ error: "Not found" })
+    }
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+})
+
+app.get('/winners', async (req, res) => {
+  const { nominee, film, page, perPage } = req.query
+  try {
+    const winners = await Nominee.aggregate([
+      {
+        $match: {
+          win: true,
+          nominee: {
+            $regex: new RegExp(nominee || "", "i")
+          },
+          film: {
+            $regex: new RegExp(film || "", "i")
+          }
+        }
+      },
+      {
+        $project: {
+          _id: 0,
+          nominee: 1,
+          film: 1,
+          year_award: 1,
+          category: 1
+        }
+      },
+      {
+        $skip: Number((page - 1) * perPage + 1)
+      },
+      {
+        $limit: Number(perPage)
```

Comment:
error 15958 says that "limit must be positive"  and you didn't provide the default value for the`perPage`, so if your user does not provide it you are not covered here 😱